server/actions/events.js

```diff
@@ -16,15 +16,41 @@ export async function createEvent(newEventData, next) {
   });
 }
 
-export async function getEvents(next) {
+export async function getEvents(startDate, endDate, next) {
 
   await dbConnect();
 
-  return EventData.find({})
-  .then((events) => {
-    return events;
-  })
-  .catch(next);
+  if (!startDate && !endDate){
+    return EventData.find({})
+    .then((events) => {
+      return events;
+    })
+    .catch(next);
+  } else if (!startDate) {
+    return EventData.find({ date: { $lte: endDate}})
+      .then((events) => {
+        return events;
+      })
+      .catch(next);
+  } else if (!endDate) {
+    return EventData.find({ date: { $gte: startDate}})
+      .then((events) => {
+        return events;
+      })
+      .catch(next);
+  } else {
+    startDate = new Date(startDate);
```

Comment:
@jwu6492 I missed this while merging. Would we need to do this parsing step for the two conditions above where only either is available? If yes, can you make another quick PR for that?